woocommerce/woocommerce
# Use a modified version of edit-site lock-unlock module to make it CYS…

… compatible with both WP 6.4 and 6.3

### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.



### Changes proposed in this Pull Request:





Closes #40850 .

Slack thread: p1697641436844019-slack-C01SFMVEYAK

WordPress 6.4 ships with a new version of `@wordpress/private-apis`, which has a slightly different consent string. The Customize Your Store task uses [edit-site](https://github.com/WordPress/gutenberg/tree/trunk/packages/edit-site) 5.15.0 package from Gutenberg, and it is no longer compatible with WordPress 6.4 due to the difference in the consent string. We can upgrade `edit-site` to the latest to fix the problem, but Customize Your Store won't work with WordPress 6.3 and below.

To make `edit-site` 5.15.0 compatible with both WordPress 6.4 and earlier versions, this PR replaces the original `lock-unlock` module with a modified version. 

I have to admit that this is quite hacky, but I couldn't find an alternative solution since the upstream change is a breaking change. Any suggestions would be appreciated.🙏 



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Create a new JN site with this branch (using WooCommerce Beta Tester). Make sure to select `Include WordPress Beta Tester` under the `Plugins` section. 
2. Complete the core profiler and make sure to connect to Jetpack.
3. Go to `Tools -> WCA Test Helper -> Features` and enable `customize-store`
4. Go to `WooCommerce -> Home` and click on `Customize your store` task.
5. Go through the steps and make sure everything displays without an error.
6. Let's test with WP 6.4
7. Go to `Tools -> Beta Testing`
8. Choose `Bleeding edge` and `Beta/RC Only` then click on `Save Changes`
9. Go to `Dashboard -> Updates`
10. Choose 'Update to version 6.4-RC1` (it can be just 6.4 or RC2, depending on when you see this step)
11. Start Customize your store task again and repeat the test.





### Changelog entry




<details>

#### Significance




#### Type




#### Message 

Replace the original lock-unlock module with a modified version to make Customize your store task compatible with both WP 6.4 and earlier versions.

#### Comment 

</details>
